src/System.Runtime/tests/System/GCTests.netstandard1.7.cs

```diff
@@ -0,0 +1,184 @@
+            GC.Collect(currentgen, GCCollectionMode.Forced, true, true);
+
+            Assert.Throws<ArgumentNullException>(() => GC.GetGeneration(wkref));
+
+        }
+        [Fact]
+        public static void GCNotifiicationNEgTests()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(-1, -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(100, -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(-1, 100));
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(10, -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(-1, 10));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(100, 10));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.RegisterForFullGCNotification(10, 100));
+
+
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.WaitForFullGCApproach(-2));
+            Assert.Throws<ArgumentOutOfRangeException>(() => GC.WaitForFullGCComplete(-2));
+        }
+
+        [Fact]
+        public static void GCNotifiicationTests()
+        {
+            Assert.True(TestWait(true, -1));
+            Assert.True(TestWait(false, -1));
+            Assert.True(TestWait(true, 0));
+            Assert.True(TestWait(false, 0));
+            Assert.True(TestWait(true, 1000));
+            Assert.True(TestWait(false, 1000));
+            Assert.True(TestWait(true, int.MaxValue));
+            Assert.True(TestWait(false, int.MaxValue));
+
+        }
+        //    [Fact]
+        public static void EndNoGCRegionTest()
+        {
+            Assert.Throws<InvalidOperationException>(() => GC.EndNoGCRegion());
+
+            Assert.True(GC.TryStartNoGCRegion(1024, true));
+            Assert.Equal(GCSettings.LatencyMode, GCLatencyMode.NoGCRegion);
+            GCSettings.LatencyMode = GCLatencyMode.LowLatency;
+            Assert.Throws<InvalidOperationException>(() => GC.EndNoGCRegion());
+
+        }
+        [Fact]
+        public static void TryStartNoGCRegionTest()
+        {
+
+            Assert.True(GC.TryStartNoGCRegion(1024));
+            Assert.Equal(GCSettings.LatencyMode, GCLatencyMode.NoGCRegion);
+            GC.EndNoGCRegion();
+
+            Assert.True(GC.TryStartNoGCRegion(1024, true));
+            Assert.Equal(GCSettings.LatencyMode, GCLatencyMode.NoGCRegion);
+            GC.EndNoGCRegion();
+
+            Assert.True(GC.TryStartNoGCRegion(1024, 1024));
+            Assert.Equal(GCSettings.LatencyMode, GCLatencyMode.NoGCRegion);
+            GC.EndNoGCRegion();
+
+            Assert.True(GC.TryStartNoGCRegion(1024, 1024, true));
+            Assert.Equal(GCSettings.LatencyMode, GCLatencyMode.NoGCRegion);
+            GC.EndNoGCRegion();
+
+        }
+        [Fact]
+        public static void TryStartNoGCRegionNegTest()
```

Comment:
Maybe we could also have test that passes a size greater than the ephemeral segment size? it's supposed to throw an `ArgumentOutOfRangeException` in that case.